inc/compatibility/fixes/class-setserverportfix.php

```diff
@@ -0,0 +1,45 @@
+<?php
+/**
+ * Set Server Port Fix
+ *
+ * This fix sets the server port based on the HTTP_USER_AGENT_HTTPS header.
+ *
+ * @package Pantheon\Compatibility\Fixes
+ */
+
+namespace Pantheon\Compatibility\Fixes;
+
+/**
+ * Class SetServerPortFix
+ *
+ * @package Pantheon\Compatibility\Fixes
+ */
+class SetServerPortFix {
+
+
+
+	public static function apply() {
+		if ( ! isset( $_SERVER['HTTP_HOST'] ) ) {
+			return;
+		}
+
+		$_SERVER['SERVER_NAME'] = $_SERVER['HTTP_HOST'];
+
+		if ( isset( $_ENV['PANTHEON_ENVIRONMENT'] ) ) {
```

Comment:
Unnecessary check. If `! isset( $_ENV['PANTHEON_ENVIRONMENT'] )` none of this should load in the first place. Which is one reason why the autoloader should be inside here: https://github.com/pantheon-systems/pantheon-mu-plugin/blob/main/pantheon.php#L15